packages/core/test/insured-bridge/e2e/helpers/MultiProviderHelper.js

```diff
@@ -0,0 +1,67 @@
+const hre = require("hardhat");
+
+const fs = require("fs");
+const path = require("path");
+
+function getProviders() {
+  return {
+    l1RpcProvider: new hre.ethers.providers.JsonRpcProvider("http://localhost:9545"),
+    l2RpcProvider: new hre.ethers.providers.JsonRpcProvider("http://localhost:8545"),
+  };
+}
+
+function getAllFiles(dirPath, arrayOfFiles = []) {
+  const files = fs.readdirSync(dirPath);
+
+  arrayOfFiles = arrayOfFiles || [];
+
+  files.forEach((file) => {
+    if (fs.statSync(dirPath + "/" + file).isDirectory()) arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles);
+    else arrayOfFiles.push(path.join(dirPath, "/", file));
+  });
+
+  return arrayOfFiles;
+}
+
+function findArtifactFromPaths(contractName, artifactsPath) {
+  const allArtifactsPaths = getAllFiles(artifactsPath);
+  const desiredArtifactPaths = allArtifactsPaths.filter((a) => a.endsWith(`/${contractName}.json`));
+
+  if (desiredArtifactPaths.length !== 1)
+    throw new Error(`Couldn't find desired artifact or found too many for ${contractName}`);
+  return JSON.parse(fs.readFileSync(desiredArtifactPaths[0], "utf-8"));
+}
+
+function getOptimismArtifact(contractName, ovm = false) {
+  const artifactsPath = ovm
+    ? path.resolve(__dirname, "../../../../../../node_modules/@eth-optimism/contracts/artifacts-ovm")
+    : path.resolve(__dirname, "../../../../../../node_modules/@eth-optimism/contracts/artifacts");
```

Comment:
Hardcoded module paths can be dangerous as yarn assumes that node is smart about resolving up the directory tree. Thoughts on using `require.resolve("@eth-optimism/package.json")` to ensure you're using that same resolution strategy?

Reply:
good point. I'll see what I can do to avoid this.